var/spack/repos/builtin/packages/googletest/package.py

```diff
@@ -9,9 +9,12 @@
 class Googletest(CMakePackage):
     """Google test framework for C++.  Also called gtest."""
     homepage = "https://github.com/google/googletest"
-    url      = "https://github.com/google/googletest/tarball/release-1.10.0"
     git      = "https://github.com/google/googletest"
 
+    def url_for_version(self, version):
+        url = "https://github.com/google/googletest/archive/refs/tags/release-{0}.tar.gz"
+        return url.format(version)
+
```

Comment:
What I meant was that setting
```python
url = "https://github.com/google/googletest/archive/refs/tags/release-1.10.0.tar.gz"
```

should allow Spack to figure out the schema for downloading new versions.

Also, why write it as `archive/refs/tags` rather than
```python
url = "https://github.com/google/googletest/archive/release-1.10.0.tar.gz"
```

I guess I fundamentally don't understand what the purpose of this PR is: what's broken? Why is this change for the better? Those are good things to describe as part of the PR.

Reply:
Interesting, I didn't know Spack can infer the versioning that way!
This PR is mainly to fix the following error message (copypasted from Slack #general channel):
```console
$ spack install googletest
==> No binary for googletest-1.10.0-5krwqhu3frbgp2euk2myn2ohg3mmyfa4 found: installing from source
==> Fetching https://github.com/google/googletest/tarball/release-1.10.0
==> Error: CommandNotFoundError: Cannot extract archive, unrecognized file extension: 'None'
```

Reply:
Thanks! I think @tmdelellis you hit this recently too.

Reply:
Ok @wyphan you're getting close: the `url_for_version` should be removed. This PR should literally just be to change the line with `url` on it. It looks like you accidentally killed a quotation mark.

Reply:
Whoops, my bad. Lemme fix that real quick.